File: Interview Street/TwitchTV/matching.cpp
#include <iostream>
#include <map>
#include <algorithm>
#include <cstdio>

using namespace std;

const int MOD = 1000000007;

int main() {
  int n;
  cin >> n;

  double ret = 1.0f;

  for (int k = 2; k <= n; ++k)
    ret = (ret * (double)(n+k) / (double)(k)) % MOD;

  printf("%lf\n", ret);
  return 0;
}

File: Interview Street/TwitchTV/matching.py
import sys
from math import factorial

MOD = 1000000007

n = int(sys.stdin.readline())

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

a = 1
b = 1
ret = 1
for k in xrange(2, n+1):
    a *= n + k
    if a % k == 0:
        a /= k
    else:
        b *= k

print (a/b) % MOD
